superset/commands/dataset/update.py

```diff
@@ -139,6 +140,13 @@ def _validate_columns(
                 ):
                     exceptions.append(DatasetColumnsExistsValidationError())
 
+            # validate python_date_format is ISO8601 format
+            for col in columns:
+                if not DatasetDAO.validate_column_pdf_is_iso8601(
+                    col.get("python_date_format", None)
```

Comment:
I'm surprised Mypy doesn't complain here given you potentially passing `None`. Personally I think it's best not to accept `None` as a valid input if you then simply return. This case should be handled separately.